### PR TITLE
Bug 1475944 Removed debug script and added diagnostics command

### PR DESCRIPTION
**Pull request by bfallonf**
For: https://bugzilla.redhat.com/show_bug.cgi?id=1475944

@sosiouxme @danwinship I removed both scripts and added the command (`oadm` not `oc adm`). Any more thoughts? Anything more needed?

### CODE REVIEW

**Comment by danwinship** [comment]
@pravisankar should we say anything more here?

File: admin_guide/sdn_troubleshooting.adoc [hunk 1]
```diff
@@ -578,14 +578,11 @@ replaced with an `iptables`-only solution.
...
 
-. Run the script at
-https://raw.githubusercontent.com/openshift/openshift-sdn/master/hack/debug.sh
-on the master (or from another machine with access to the master) to
-generate useful debugging information.
-. When debugging IP failover problems, run the script at
-https://raw.githubusercontent.com/openshift/openshift-sdn/master/hack/ipf-debug.sh
-on the master (or from another machine with access to the master) to
-generate useful debugging information.
+Run the following to diagnose common network issues:
```
Comment (sosiouxme) [inline admin_guide/sdn_troubleshooting.adoc]: It's probably worth saying in what context to run this (basically that you have to run it with the `admin.kubeconfig`). Or possibly a link to more general instructions for running the diagnostics.

**Review by sosiouxme — COMMENTED** [review]
s/disagnostics/diagnostics/ in the title :)

**Comment by bfallonf** [comment]
Thanks, all. @sosiouxme I updated it to "As a cluster administrator, run the diagnostics tool to diagnose common network issues". Is that what you mean by the "admin.kubeconfig" file? Also added a spiel about the tool with a link to the section in the Admin Guide.

@pravisankar please let me know if anything more needs to be added.

**Comment by pravisankar** [comment]
@bfallonf 
Yes, 'odm diagnostics NetworkCheck' is the replacement for debug.sh script but this script can be useful in one or two cases.
Both diagnostics tool and debug.sh script depends on openshift master to be up and running so that it gets responses for 'oc get <resource-name>' queries. Unlike debug.sh script, diagnostics cmd doesn't need ssh access and creates test namespaces/pods to figure out various network issues but if the master has a problem with scheduling, test pods may not be deployed and the tool doesn't give any useful information. debug.sh script may provide some useful info in this case as it operates on existing pods in the cluster. When we have scheduling problem, 'oadm diagnostics NetworkCheck' may not be useful but 'oadm diagnostics NodeConfig/MasterConfig' could be useful. Since this script is not officially supported, not regularly tested on the recent releases and may be helpful in a very few cases. I vote for getting rid of it from the docs.

IP failover debug script (ipf-debug.sh) functionality is not included in the diagnostics tool yet, so this script can be useful. Created trello card: https://trello.com/c/JCC3l2qc 
Until this card is done, can we keep ipf-debug.sh in the docs but say this is unsupported?

Definitely we need an update to 'Diagnostics Tool section'(diagnostics_tool.adoc), the tool can do more things than what's currently documented.

**Comment by bfallonf** [comment]
@pravisankar Sure. Thanks for that info. I added a note box with the IP failover script. I think that's the best bet until something is done about the card you created.

Not too sure what to say about the potential for scheduling errors. It might be better to improve the diagnostic tool info rather than talking about how to diagnose the issues here. Please let me know if there's more you'd like done with this PR.

**Comment by pravisankar** [comment]
@bfallonf  changes LGTM

**Comment by bfallonf** [comment]
Thanks @pravisankar . @openshift/team-documentation PTAL

File: admin_guide/sdn_troubleshooting.adoc [hunk 2]
```diff
@@ -576,16 +576,29 @@ NAT'd to the right port on the local machine for the `kubeproxy`.
...
+As a cluster administrator, run the diagnostics tool to diagnose common network
+issues:
+
+----
+# oadm diagnostics NetworkCheck
+----
+
+The diagnostics tool runs a series of checks for error conditions for the
+specified component. See the
+/admin_guide/diagnostics_tool.adoc#overview[Diagnostics Tool section] for more
```
Comment (mburke5678) [inline admin_guide/sdn_troubleshooting.adoc]: Does this cross reference need the `xref:..` before it? Or, is this another format that works just as well?

File: admin_guide/sdn_troubleshooting.adoc [hunk 3]
```diff
@@ -576,16 +576,29 @@ NAT'd to the right port on the local machine for the `kubeproxy`.
...
+information.
+
+[NOTE]
+====
+Currently, the diagnostics tool cannot determine information about IP failover
```
Comment (mburke5678) [inline admin_guide/sdn_troubleshooting.adoc]: s/determine information about/diagnose (?)

**Comment by mburke5678** [comment]
@bfallonf Couple of nits. Otherwise LGTM

File: admin_guide/sdn_troubleshooting.adoc (hunk 2 above)
Comment (bfallonf) [inline admin_guide/sdn_troubleshooting.adoc]: Oops. Added.

File: admin_guide/sdn_troubleshooting.adoc (hunk 3 above)
Comment (bfallonf) [inline admin_guide/sdn_troubleshooting.adoc]: Hmm right. I didn't like that cos it says diagnostics tool a few words before, but it's better. Changed.

**Comment by bfallonf** [comment]
Thanks @mburke5678 I'll merge.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../admin_guide/sdn_troubleshooting.adoc#admin-guide-sdn-troubleshooting[Troubleshooting OpenShift SDN]
|Changed the Further Help section to xref:../admin_guide/sdn_troubleshooting.adoc#sdn-troubleshooting-diagnostics-tool[Finding Network Issues Using the Diagnostics Tool] and added information about the Diagnostic Tool.
%

**Pull request merged**